### PR TITLE
fix: adapt setup scripts to run properly on distrobox

### PR DESCRIPTION
**Pull request by imobachgs**
Due to a problem with [ruby-augeas](https://rubygems.org/gems/augeas) and Ruby 3.3, the setup script does not work anymore. In our continuous integration we decided to use the system libraries, which causes bundler to install software into `/usr/lib`. Of course, we do not want that behavior everywhere.

Alternatively, this PR adapts the script to work using distrobox. The setup is rather simple:

```
distrobox create --root --init --image tumbleweed --name agama
distrobox enter --root agama
./setup.sh
```

The script detects that it is running on distrobox and it workarounds the augeas problem like in our continuous integration.

Please, see #1685 for more context.

### CODE REVIEW

**Comment by dgdavid** [comment]
As said in IRC, it looks good to me. What is more, I was able to run the project in a similar way by using distrobox (thanks for the help). But I guess this distrobox thingie should be document as well since there could be people like me that have no idea about such a tool.

> Please, see https://github.com/agama-project/agama/pull/1687 for more context.

Do you mean https://github.com/agama-project/agama/issues/1685?

**Comment by imobachgs** [comment]
JFYI, I still fix a problem with NetworkManager.

**Comment by imobachgs** [comment]
I have adapted the script to use the host's D-Bus system socket when running on Distrobox.

File: setup-services.sh [hunk 1]
```diff
@@ -116,14 +117,15 @@ fi
       sed -e '/gemspec/a gem "ruby-dbus", path: "/checkout-ruby-dbus"' -i Gemfile
   fi
 
-  if [ -n "$CI" ]; then
+  if [ -n "${CI:-}" ]; then
     # in CI reuse the pre-installed system gems from RPMs
     bundle config set --local disable_shared_gems 0
+    sudo bundle install
```
Comment (jreidinger) [inline setup-services.sh:123]: it should use `$SUDO` see https://github.com/agama-project/agama/pull/1687/files#diff-327a0d405c7c05ecb8395935997d54a646315e7da5e275748c565467fc3b7f61R17

File: setup-services.sh [hunk 2]
```diff
@@ -147,7 +150,7 @@ $SUDO $ZYPPER install \
   cd $MYDIR/rust
   cargo build
 
-  ln -st /usr/bin $MYDIR/rust/target/debug/agama{,*server}
+  sudo ln -sft /usr/bin $MYDIR/rust/target/debug/agama{,*server}
```
Comment (jreidinger) [inline setup-services.sh:153]: and same here `https://github.com/agama-project/agama/pull/1687/files#diff-327a0d405c7c05ecb8395935997d54a646315e7da5e275748c565467fc3b7f61R17` to allow to use different way to have root permission

**Review by jreidinger — APPROVED** [review]

**Pull request merged**